[cutlass backend] Minor lru_cache to slightly speed up filtering ops

henrylhtsang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #152577

For default level, it went from 0.11332 seconds to Filtering took 0.10064 seconds.

You can't really apply lru_cache too aggressively. For example, hashing a cutlass op takes a long time.

Removing a log further bring it down to 0.07202 seconds

Differential Revision: [D73971021](https://our.internmc.facebook.com/intern/diff/D73971021/)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

henrylhtsang committed:
[cutlass backend] Minor lru_cache to slightly speed up filtering ops

For default level, it went from 0.11332 seconds to Filtering took 0.10064 seconds.

You can't really apply lru_cache too aggressively. For example, hashing a cutlass op takes a long time.

Removing a log further bring it down to 0.07202 seconds

Differential Revision: [D73971021](https://our.internmc.facebook.com/intern/diff/D73971021/)

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] Minor lru_cache to slightly speed up filtering ops"


For default level, it went from 0.11332 seconds to Filtering took 0.10064 seconds.

You can't really apply lru_cache too aggressively. For example, hashing a cutlass op takes a long time.

Removing a log further bring it down to 0.07202 seconds

Differential Revision: [D73971021](https://our.internmc.facebook.com/intern/diff/D73971021/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang commented:
@pytorchbot merge

wdvr commented:
@pytorchmergebot revert -m "failing test_unary_ufuncs.py::TestUnaryUfuncsCUDA::test_reference_numerics_large_jiterator_unary_cuda_complex64 [GH job link](https://github.com/pytorch/pytorch/actions/runs/14787347116/job/41519095088) [HUD commit link](https://hud.pytorch.org/pytorch/pytorch/commit/1fef3cdabc3f79fd0cbf9273052057ef6122710f)" -c nosignal

henrylhtsang commented:
@wdvr if I want to re-land, what label can I get to trigger the ci that was broken by this PR?

henrylhtsang commented:
@pytorchbot rebase -b viable/strict

henrylhtsang committed:
Update on "[cutlass backend] Minor lru_cache to slightly speed up filtering ops"


For default level, it went from 0.11332 seconds to Filtering took 0.10064 seconds.

You can't really apply lru_cache too aggressively. For example, hashing a cutlass op takes a long time.

Removing a log further bring it down to 0.07202 seconds

Differential Revision: [D73971021](https://our.internmc.facebook.com/intern/diff/D73971021/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

wdvr commented:
@henrylhtsang ci-no-td is what you’re looking for, that one skips target determination and runs all pull test. A revert auto adds it so you should be good. Look for 

linux-focal-cuda12.6-py3.10-gcc11 / test

in the test section, that one should run all now and you should see TestUnaryUfuncsCUDA in the output

henrylhtsang commented:
@pytorchbot merge
